controller/appcontroller.go

```diff
@@ -212,6 +211,18 @@ func (ctrl *ApplicationController) setAppManagedResources(a *appv1.Application,
 	return tree, ctrl.cache.SetAppManagedResources(a.Name, managedResources)
 }
 
+func isOrphanedResource(proj *appv1.AppProject, key kube.ResourceKey) bool {
```

Comment:
I think you should split this funcn and remove the permission check at the end, push that back into the calling func

Reply:
done